charts/reporting-operator/templates/reporting-operator-deployment.yaml

```diff
@@ -156,12 +156,21 @@ spec:
               name: reporting-operator-config
               key: prometheus-datasource-import-from
               optional: true
+{{- /* neither specified = no auth used; both specified = error; either = correct & authenticated */ -}}
+{{- if .Values.spec.config.prometheusImporter.auth.tokenSecret.enabled }}  {{- if .Values.spec.config.prometheusImporter.auth.useServiceAccountToken  }} {{ required "cannot use both token from secret and token from service account" null }} {{- end}} {{- end }}
         - name: REPORTING_OPERATOR_PROMETHEUS_BEARER_TOKEN_FILE
           valueFrom:
             configMapKeyRef:
               name: reporting-operator-config
               key: prometheus-bearer-token-file
               optional: true
+{{- if .Values.spec.config.prometheusImporter.auth.tokenSecret.enabled }}
```

Comment:
We decided on use a volumeMount for `tokenSecret` so this env var doesn't need to be set, instead we should be mounting the file and setting `prometheus-bearer-token-file` in the configmap to the location of the file